helm/cluster-cloud-director/templates/helpers/_workers.tpl

```diff
@@ -3,11 +3,15 @@ Generates template spec for worker machines.
 */}}
 {{- define "worker-vcdmachinetemplate-spec" -}}
 {{- $pool := $.nodePool.config | deepCopy -}}
+{{- $pool = set $pool "diskSize" ( include "calculateDiskBytes" $pool.diskSizeGB ) -}}
+{{- $pool = unset $pool "diskSizeGB" -}}
 {{- $pool = unset $pool "replicas" -}}
 {{- $pool = unset $pool "machineHealthCheck" -}}
+{{- $pool = unset $pool "customNodeLabels" -}}
+{{- $pool = unset $pool "customNodeTaints" -}}
```

Comment:
these two aren't usually set but if they are then they leak into the vcdmachinetemplate spec so we need to delete them (does not affect the control plane)